src/Connection/TermObjects.php

```diff
@@ -52,179 +49,162 @@ public static function register_connections() {
+					'oneToOne'           => true,
+					'resolve'            => function ( Term $term, $args, AppContext $context, $info ) use ( $tax_object ) {
+						if ( ! isset( $term->parentDatabaseId ) || empty( $term->parentDatabaseId ) ) {
+							return null;
+						}
 
-				/** @var \WP_Post_Type $post_type_object */
-				$post_type_object = get_post_type_object( $allowed_post_type );
+						$resolver = new TermObjectConnectionResolver( $term, $args, $context, $info, $tax_object->name );
+						$resolver->set_query_arg( 'include', $term->parentDatabaseId );
 
-				if ( empty( get_object_taxonomies( $allowed_post_type ) ) ) {
-					continue;
-				}
+						return $resolver->one_to_one()->get_connection();
+					},
+				] );
 
 				register_graphql_connection( [
-					'fromType'       => $post_type_object->graphql_single_name,
-					'toType'         => 'TermNode',
-					'fromFieldName'  => 'terms',
-					'queryClass'     => 'WP_Term_Query',
-					'connectionArgs' => self::get_connection_args(
-						[
-							'taxonomies' => [
-								'type'        => [ 'list_of' => 'TaxonomyEnum' ],
-								'description' => __( 'The Taxonomy to filter terms by', 'wp-graphql' ),
-							],
-						]
-					),
-					'resolve'        => function ( Post $post, $args, AppContext $context, ResolveInfo $info ) {
-						$taxonomies = \WPGraphQL::get_allowed_taxonomies();
-						$terms      = wp_get_post_terms( $post->ID, $taxonomies, [ 'fields' => 'ids' ] );
-						if ( empty( $terms ) || is_wp_error( $terms ) ) {
+					'fromType'           => $tax_object->graphql_single_name,
+					'toType'             => $tax_object->graphql_single_name,
+					'fromFieldName'      => 'ancestors',
+					'description'        => __( 'The ancestors of the node. Default ordered as lowest (closest to the child) to highest (closest to the root).', 'wp-graphql' ),
+					'connectionTypeName' => ucfirst( $tax_object->graphql_single_name ) . 'ToAncestors' . ucfirst( $tax_object->graphql_single_name ) . 'Connection',
+					'resolve'            => function ( Term $term, $args, AppContext $context, $info ) use ( $tax_object ) {
+						if ( ! $tax_object instanceof \WP_Taxonomy ) {
 							return null;
 						}
-						$resolver = new TermObjectConnectionResolver( $post, $args, $context, $info, $taxonomies );
-						$resolver->set_query_arg( 'include', $terms );
 
-						return $resolver->get_connection();
+						$ancestor_ids = get_ancestors( absint( $term->term_id ), $term->taxonomyName, 'taxonomy' );
+
+						if ( empty( $ancestor_ids ) ) {
+							return null;
+						}
 
+						$resolver = new TermObjectConnectionResolver( $term, $args, $context, $info, $tax_object->name );
+						$resolver->set_query_arg( 'include', $ancestor_ids );
+
+						return $resolver->get_connection();
 					},
 				] );
 			}
 		}
 
+		// Register a connection from each post type that
+		foreach ( $allowed_post_types as $post_type_object ) {
```

Comment:
I'm unclear why this foreach loop is separate from earlier one on L85.

Reply:
ah yes, so the one on line 85 is within the `foreach ( $allowed_taxonomies as $tax_object ) {` loop. 

So, this is getting the intersection of allowed post types for each allowed taxonomy so we can create connections:

- from TaxonomyA -> PostTypeA, etc

This foreach, on line 176 is top level post types, not nested within another foreach. 

I'm open to re-thinking how this was done if you have ideas to dry it up or whatever.